x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -333,29 +321,15 @@ export const signalRulesAlertType = ({
             buildRuleMessage,
           });
 
-          previousSignals.aggregations.threshold.buckets.forEach((bucket: ThresholdQueryBucket) => {
-            esFilter.bool.filter.push(({
-              bool: {
-                must_not: {
-                  bool: {
-                    must: [
-                      {
-                        term: {
-                          [threshold.field || 'signal.rule.rule_id']: bucket.key,
-                        },
-                      },
-                      {
-                        range: {
-                          [timestampOverride ?? '@timestamp']: {
-                            lte: bucket.lastSignalTimestamp.value_as_string,
-                          },
-                        },
-                      },
-                    ],
-                  },
-                },
-              },
-            } as unknown) as Filter);
+          const esFilter = await getFilter({
+            type,
+            filters: filters?.concat(bucketFilters),
```

Comment:
`filters?.concat(bucketFilters)` returns `undefined` if `filters` is undefined. Looks like `filters ? filters.concat(bucketFilters) : bucketFilters` would give the intended result? `filters` defaults to `[]` if not provided in the UI but it's optional in the API so it could be `undefined` here.
```suggestion
            filters: filters ? filters.concat(bucketFilters) : bucketFilters,
```